cdap-data-fabric/src/main/java/io/cdap/cdap/spi/metadata/dataset/SearchHelper.java

```diff
@@ -304,22 +312,83 @@ private MetadataSearchResponse search(Set<MetadataScope> scopes, SearchRequest r
       finalResults, cursors, request.shouldShowHidden(), request.getEntityScopes());
   }
 
-  private Set<MetadataEntity> getSortedEntities(List<MetadataEntry> results, SortInfo sortInfo) {
+  private Map<MetadataEntity, Map<String, Integer>> hashResults(List<MetadataResultEntry> results) {
+    Map<MetadataEntity, Map<String, Integer>> hashedResults = new LinkedHashMap<>();
+    for (MetadataResultEntry m : results) {
+      MetadataEntity entity = m.getMetadataEntity();
+      String label = m.getLabel();
+
+      if(!hashedResults.containsKey(entity)){
```

Comment:
Missing space between `if` and `(`. The checkstyle build will fail.

Reply:
it did, thank you!